agent/fw_drupal8.c

```diff
@@ -20,6 +21,60 @@
 
 #define PHP_PACKAGE_NAME "drupal/core"
 
+NR_PHP_WRAPPER(nr_drupal_exception) {
+  int priority = nr_php_error_get_priority(E_ERROR);
+  zval* event = NULL;
+  zval* exception = NULL;
+
+  /* Warning avoidance */
+  (void)wraprec;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL8);
+
+  if (NR_SUCCESS != nr_txn_record_error_worthy(NRPRG(txn), priority)) {
+    NR_PHP_WRAPPER_CALL;
+    goto end;
+  }
+
+  /* Get the event that was given. */
+  event = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+
+  /* Call the original function. */
+  NR_PHP_WRAPPER_CALL;
+
+  if (0 == nr_php_is_zval_valid_object(event)) {
+    nrl_verbosedebug(NRL_TXN,
+                     "Drupal: ExceptionSubscriber::onException() does not "
+                     "have an `event` parameter");
+    goto end;
+  }
+
+  /*
+   * Get the exception from the event.
+   */
```

Comment:
This code seems to rely on `nr_php_call()` gracefully handling if these methods did not exist.  Have you verified this is how it works?
I am also curious if these calls could return a valid zval that is not an exception object and perhaps it would be best to verify it is what we expect?
Or does `nr_php_error_record_exception()` handle this inspection and verification?

Reply:
> This code seems to rely on nr_php_call() gracefully handling if these methods did not exist. Have you verified this is how it works?

I believe this is outside the scope of this PR. We use `nr_php_call` in this same manner throughout the agent, but for what it's worth, I do believe it handles the case that it does not exist gracefully.

> I am also curious if these calls could return a valid zval that is not an exception object and perhaps it would be best to verify it is what we expect?

The very first call in `nr_php_error_record_exception` is to `nr_php_error_zval_is_exception`.